pkg/arvo/app/group-hook.hoon

```diff
@@ -0,0 +1,270 @@
+  ?~  pax
+    [[ost.bol %quit ~]~ this]
+  ?.  (~(has by synced) pax)
+    [[ost.bol %quit ~]~ this]
+  =/  grp=(unit group)  (group-scry pax)
+  ?^  grp
+    :_  this
+    [ost.bol %diff [%group-update [%path u.grp pax]]]~
+  [[ost.bol %quit ~]~ this]
+::
+++  diff-group-update
+  |=  [wir=wire diff=group-update]
+  ^-  (quip move _this)
+  ?:  =(src.bol our.bol)
+    (handle-local diff)
+  (handle-foreign diff)
+::
+++  handle-local
+  |=  diff=group-update
+  ^-  (quip move _this)
+  ?-  -.diff
+      %keys
+    [~ this]
+      %path
+    [~ this]
+      %bundle
+    [~ this]
+  ::
+      %add
+    :_  this
+    %+  turn  (prey:pubsub:userlib [%group pax.diff] bol)
+    |=  [=bone *]
+    ^-  move
+    [bone %diff [%group-update diff]]
+  ::
+      %remove
+    :_  this
+    %+  turn  (prey:pubsub:userlib [%group pax.diff] bol)
+    |=  [=bone *]
+    ^-  move
+    [bone %diff [%group-update diff]]
+  ::
+      %unbundle
+    :_  this(synced (~(del by synced) pax.diff))
+    %+  weld
+    ^-  (list move)
+    %+  turn  (prey:pubsub:userlib [%group pax.diff] bol)
+    |=  [=bone *]
+    ^-  move
+    [bone %diff [%group-update diff]]
+    ^-  (list move)
+    %+  turn  (prey:pubsub:userlib [%group pax.diff] bol)
+    |=  [=bone *]
+    ^-  move
+    [bone %quit ~]
+  ::
+  ==
+::
+++  handle-foreign
+  |=  diff=group-update
+  ^-  (quip move _this)
+  ?-  -.diff
+      %keys
+    [~ this]
+      %bundle
+    [~ this]
+  ::
+      %path
+    ?~  pax.diff
+      [~ this]
+    =/  ship  (~(get by synced) pax.diff)
+    ?~  ship
+      [~ this]
+    ?.  =(src.bol u.ship)
+      [~ this]
+    :_  this
+    :~  (group-poke pax.diff [%unbundle pax.diff])
+        (group-poke pax.diff [%bundle pax.diff])
+        (group-poke pax.diff [%add members.diff pax.diff])
+    ==
+  ::
+      %add
+    ?~  pax.diff
+      [~ this]
+    =/  ship  (~(get by synced) pax.diff)
+    ?~  ship
+      [~ this]
+    ?.  =(src.bol u.ship)
+      [~ this]
+    :_  this
+    :~  (group-poke pax.diff diff)
+    ==
+  ::
+      %remove
+    ?~  pax.diff
+      [~ this]
+    =/  ship  (~(get by synced) pax.diff)
+    ?~  ship
+      [~ this]
+    ?.  =(src.bol u.ship)
+      [~ this]
+    :_  this
+    :~  (group-poke pax.diff diff)
+    ==
+  ::
+      %unbundle
+    ?~  pax.diff
+      [~ this]
+    =/  ship  (~(get by synced) pax.diff)
+    ?~  ship
+      [~ this]
+    ?.  =(src.bol u.ship)
+      [~ this]
+    :_  this(synced (~(del by synced) pax.diff))
+    :~  (group-poke pax.diff diff)
+    ==
+  ::
+  ==
+::
+++  quit
+  |=  wir=wire
+  ^-  (quip move _this)
+  =/  wir  `(list @tas)`wir
+  =/  =ship  (slav %p &1:wir)
+  =.  wir  ?^  wir  t.wir  ~
+  =.  wir  ?^  wir  t.wir  ~
+  ?:  (~(has by synced) wir)
+    =/  group-path  [%group wir]
+    =/  group-wire  [(scot %p ship) group-path]
+    :_  (track-bone group-wire)
+    [ost.bol %peer group-wire [ship %group-hook] group-path]~
+  ::  no-op
+  [~ this]
+::
+++  reap
+  |=  [wir=wire saw=(unit tang)]
```

Comment:
If you don't expect this to fail often, might not be the worst idea to `+slog` out the error.

Reply:
I expect it to fail whenever a person's permissions are revoked. Not sure if that counts as often?

Reply:
Seems like if you tried to subscribe to someone else's group and failed, you might want a message saying that it failed.  That way you might see something like "insufficient permissions" instead of it just looking like it hung.

Reply:
Added this. Thanks.